core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -297,22 +310,43 @@ private CompletableFuture<AggregatedHttpFile> loadVersions(Executor executor) {
             }, executor));
         }
 
-        private CompletableFuture<AggregatedHttpFile> loadSpecifications(Executor executor) {
-            return files.computeIfAbsent(specificationPath, key -> {
-                return CompletableFuture.supplyAsync(() -> {
-                    final DocStringSupport docStringSupport = new DocStringSupport(services);
-                    ServiceSpecification spec = generate(services);
-                    spec = docStringSupport.addDocStrings(spec);
-                    spec = exampleSupport.addExamples(spec);
-                    try {
-                        final byte[] content = jsonMapper.writerWithDefaultPrettyPrinter()
-                                                         .writeValueAsBytes(spec);
-                        return toFile(content, MediaType.JSON_UTF_8);
-                    } catch (JsonProcessingException e) {
-                        throw new RuntimeException(e);
-                    }
-                }, executor);
-            });
+        private CompletableFuture<ServiceSpecification> generateServiceSpecification(Executor executor) {
+            return CompletableFuture.supplyAsync(() -> {
+                final DocStringSupport docStringSupport = new DocStringSupport(services);
+                ServiceSpecification spec = generate(services);
+                spec = docStringSupport.addDocStrings(spec);
+                spec = exampleSupport.addExamples(spec);
+                return spec;
+            }, executor);
+        }
+
+        private CompletableFuture<AggregatedHttpFile> loadSpecifications(
+                CompletableFuture<ServiceSpecification> specificationFuture) {
+            return files.computeIfAbsent(SPECIFICATION_PATH, key -> specificationFuture.thenApply((spec) -> {
```

Comment:
It seems like I'm getting a warning from intellij

```suggestion
            return files.computeIfAbsent(SPECIFICATION_PATH, key -> specificationFuture.thenApply(spec -> {
```